realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionImpl.java

```diff
@@ -539,6 +548,7 @@ protected E convertRowToObject(UncheckedRow row) {
         if (className != null) {
             return new OrderedRealmCollectionSnapshot<E>(realm, collection, className);
         } else {
+            assert classSpec != null;
```

Comment:
Yes, that was my concern. From my understandings asserts do not work on Android or are hard to enable, but I guess I can learn to live with these in order to suppress warnings. They are no worse than comments at least. 

Could these be replaced by a comment that suppresses the warning? That would at least make the way we suppress these warnings consistent?